### PR TITLE
feat: api client modal package

### PR DESCRIPTION
**Pull request by amritk**
This PR stands up the API client modal package. Its not complete but its working! I figure we can merge this here then follow up with moving the addressbar + setting config, routes etc (ez stuff with the clientRouter)

image

test with `pnpm dev:modal`

Also this is the ESM version so it should work in any framework!

### CODE REVIEW

**Review by marclave — APPROVED** [review]
the future is here today, awesome work @amritk i can't wait to get this fully swapped.

File: packages/components/src/components/ScalarModal/ScalarModal.vue [hunk 1]
```diff
@@ -37,7 +37,7 @@ const modal = cva({
       sm: 'max-w-screen-sm',
       md: 'max-w-screen-md',
       lg: 'max-w-screen-lg',
-      full: 'mt-0',
+      full: 'mt-0 overflow-hidden',
```
Comment (antlio) [inline packages/components/src/components/ScalarModal/ScalarModal.vue:40]: what did you encounter here?

File: packages/api-client-modal/playground/main.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { createScalarClient } from '@/api-client-modal'
-import content from '@scalar/galaxy/latest.json?raw'
+import content from '@scalar/galaxy/latest.json'
```
Comment (antlio) [inline packages/api-client-modal/playground/main.ts:2]: add to add this back to test locally

Comment (amritk) [reply packages/api-client-modal/playground/main.ts:2]: hmmm will ts-ignore

**Pull request merged**